src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -42,62 +42,67 @@ static SpellFxRenderer *spell_fx_renderer = EngineIoc::ResolveSpellFxRenderer();
 const size_t CastSpellInfoCount = 10;
 std::array<CastSpellInfo, CastSpellInfoCount> pCastSpellInfo;
 
+// Common initialization of SpriteObject for spell casting
+static void
+InitSpellSprite(SpriteObject* spritePtr,
```

Comment:
This function depends on `spritePtr->uType` being set. This is somewhat non-trivial, so let's add a doxydoc.

Also, I suggest adding `assert(spritePtr && spritePtr->uType != SPRITE_NULL);` at the start of the function. I'm not sure this is always true though. Same for `assert(pCastSpell->uSpellID != 0)` – add it if it's actually true.

Regarding code formatting throughout the repo, `static void` usually goes on the same line as the function name.